### ISSUE TITLE
Recognize `block_encryption_mode` from downstream

### ISSUE THREAD

**Issue by kennytm** [pingcap/tidb-lightning#504]
Lightning as of 4.0.9 always use the default `block_encryption_mode`. Since `block_encryption_mode` affects result of `aes_encrypt()` and `aes_decrypt()`, using the wrong mode may produce incorrect result for stored generated columns.

We should always copy the `block_encryption_mode` from downstream, just like `tidb_row_format_version`.

**Comment by glorv** [comment]
I'm a little confused about this, In what condition, will lightning call the `aes_encrypt()` and `aes_decrypt`? Maybe in column default value expression?
Is there any other session or global variables like this? 🤔

**Comment by kennytm** [comment]
@glorv stored generated columns, default expression (this one seems not yet supported by parser), maybe expression index. 

(P.S. turns out Lightning doesn't support stored generated columns yet and will produce wrong result.)

**Comment by kennytm** [comment]
# System variables required by MySQL 8.0 functions

## max_allowed_packet

Affects almost all string functions, even as intermediate step. Produced NULL if exceeding `@@max_allowed_packet`.

<details><summary>Demo</summary>

```sql
mysql> select sha1(space(999999));
+------------------------------------------+
| sha1(space(999999))                      |
+------------------------------------------+
| dd20a2d11138f380cae145cd4361b2ac885e591d |
+------------------------------------------+
1 row in set (0.09 sec)

mysql> set max_allowed_packet=4096;
Query OK, 0 rows affected (0.06 sec)

mysql> select sha1(space(999999));
+---------------------+
| sha1(space(999999)) |
+---------------------+
| NULL                |
+---------------------+
1 row in set, 1 warning (0.04 sec)
```

</details>    

## div_precision_increment

Not implemented by TiDB yet (pingcap/tidb#21485), but affects the result of the `/` operator.

<details><summary>Demo</summary>

```
mysql> select 1/3, length(1/3);
+--------+-------------+
| 1/3    | length(1/3) |
+--------+-------------+
| 0.3333 |           6 |
+--------+-------------+
1 row in set (0.00 sec)

mysql> set @@div_precision_increment = 10;
Query OK, 0 rows affected (0.00 sec)

mysql> select 1/3, length(1/3);
+--------------+-------------+
| 1/3          | length(1/3) |
+--------------+-------------+
| 0.3333333333 |          12 |
+--------------+-------------+
1 row in set (0.00 sec)
```

</details>

## time_zone

Our docs does mention that Lightning and TiDB should be running in the same time zone, so should be fine whether we replicate this or not.

<details><summary>Demo</summary>

```
mysql> set time_zone='-08:00';
Query OK, 0 rows affected (0.07 sec)

mysql> select cast(from_unixtime(1) as char);
+--------------------------------+
| cast(from_unixtime(1) as char) |
+--------------------------------+
| 1969-12-31 16:00:01            |
+--------------------------------+
1 row in set (0.04 sec)

mysql> set time_zone='+00:00';
Query OK, 0 rows affected (0.05 sec)

mysql> select cast(from_unixtime(1) as char);
+--------------------------------+
| cast(from_unixtime(1) as char) |
+--------------------------------+
| 1970-01-01 00:00:01            |
+--------------------------------+
1 row in set (0.04 sec)
```

</details>

## lc_time_names

Another feature not yet implemented in TiDB. Affects the `DATE_FORMAT`, `DAYNAME` and `MONTHNAME` functions.

<details><summary>Demo</summary>

```
mysql> select date_format('2000-01-01', '%M');
+---------------------------------+
| date_format('2000-01-01', '%M') |
+---------------------------------+
| January                         |
+---------------------------------+
1 row in set (0.04 sec)

mysql> set lc_time_names = 'ja_JP';
Query OK, 0 rows affected (0.06 sec)

mysql> select date_format('2000-01-01', '%M');
+---------------------------------+
| date_format('2000-01-01', '%M') |
+---------------------------------+
| 1月                             |
+---------------------------------+
1 row in set (0.00 sec)
```

</details>

## default_week_format

Affects the `WEEK` function if the `mode` argument was not provided.

<details><summary>Demo</summary>

```
mysql> select week('2020-02-02');
+--------------------+
| week('2020-02-02') |
+--------------------+
|                  5 |
+--------------------+
1 row in set (0.05 sec)

mysql> set @@default_week_format = 4;
Query OK, 0 rows affected (0.05 sec)

mysql> select week('2020-02-02');
+--------------------+
| week('2020-02-02') |
+--------------------+
|                  6 |
+--------------------+
1 row in set (0.04 sec)
```

</details>

## regexp_stack_limit, regexp_time_limit

Not implemented by TiDB, and seems irrelevant due to regexp engine difference.

## block_encryption_mode

This issue. Affects `AES_ENCRYPT` and `AES_DECRYPT`.

## pseudo_thread_id

Affects the result of `CONNECTION_ID` function, but it makes no sense evaluating this in Lightning, and this is unsupported in TiDB.

## ~~group_concat_max_len~~

Affects `GROUP_CONCAT`, but aggregation functions are disallowed so we are fine to not copy it.
